### PR TITLE
Added documentation on what must be done to adapt volcano to the k8s version upgrade.

### PR DESCRIPTION
**Pull request by guoqinwill**
Added documentation on what must be done to adapt volcano to the k8s version upgrade.

### CODE REVIEW

File: docs/design/adapt-k8s-todo.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+![img_1.png](./images/adapt-k8s-todo.png)
+
+The main things that need to be done are as follows:
+1. First change gomod to adapt to the k8s version (and introduce the updated version of volcano apis).
+2. Upgrade the go version of go mod file and KUBE_VERSION in Dockerfile to keep consistent with k8s.
+2. Synchronize the changes in k8s regarding volumebinding, because currently volcano maintains volumebinding separately in order to be compatible with lower version kubernete's access to the volumebinding api.
+3. Other commits are to solve compilation and other problems that arise after adaptation, including API changes, scheduling policy updates, function name changes, etc. When solving, you can refer to the changelog released during the k8s upgrade.
+4. Run unit-test and ensure it passes.
+5. Run e2e verification on the specified version clusters and pass it (including the latest k8s version and historical versions).
+6. Update the Kubernetes compatibility of README file.
```
Comment (Monokaix) [inline docs/design/adapt-k8s-todo.md:26]: Please add that volcano should support kube-scheduler's new beta features,  and these features are tracked using separate issues and PRs

Comment (Monokaix) [reply docs/design/adapt-k8s-todo.md:26]: Also boldfaced to highlight, other important parts should also be boldfaced .

Comment (guoqinwill) [reply docs/design/adapt-k8s-todo.md]: done

File: docs/design/adapt-k8s-todo.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Motivation
+
+In order to ensure that volcano is compatible with the new version of kubernetes, the k8s version of volcano needs to be routinely upgraded.
+
+
+## TODO things for upgrade
+
+####  Upgrade volcano apis
+First,we need to upgrade the k8s version of volcano apis, such as:https://github.com/volcano-sh/apis/pull/124.
```
Comment (lowang-bh) [inline docs/design/adapt-k8s-todo.md:11]: A blank space is need after `First,`.

Comment (guoqinwill) [reply docs/design/adapt-k8s-todo.md]: ok

**Review by lowang-bh — COMMENTED** [review]
/lgtm

**Review by william-wang — APPROVED** [review]

**Pull request merged**